teps/0146-parameters-in-script.md

```diff
@@ -0,0 +1,605 @@
+
+### Conformance
+
+<!--
+https://github.com/tektoncd/community/blob/main/design-principles.md#conformance
+
+- Does this proposal require the user to understand how the Tekton API is implemented?
+- Does this proposal introduce additional Kubernetes concepts into the API? If so, is this necessary?
+- If the API is changing as a result of this proposal, what updates are needed to the
+[API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md)?
+-->
+- Does this proposal require the user to understand how the Tekton API is implemented?
+  - No, not directly.  The users will need to know this `.env` extension exists but this can states in our 
+  best practices and also showup in the warning/error when the user attempt to not use this extension w/ `script`.
+- Does this proposal introduce additional Kubernetes concepts into the API? If so, is this necessary?
+  - No
+- If the API is changing as a result of this proposal, what updates are needed to the
+[API spec](https://github.com/tektoncd/pipeline/blob/main/docs/api-spec.md)?
+  - None
+
+### Performance
+The proposed solution + alternatives outlined here should not meaningfully impact performance.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Tekton ecosystem. Consider including folks that also work outside the WGs
+or subproject.
+- How will security be reviewed and by whom?
+- How will UX be reviewed and by whom?
+-->
+- What are the risks of this proposal and how do we mitigate?
+  - This proposal is driven by the necessity to mitigate risks.  UX is the main risk which is the proposed
+  solution attempts to mitigate by allowing a phased wrollout (keeping prior functionality) and offering sensible
+  warnings, errors, and building on existing Tekton UX (`$(params.<name>.env)` similar to existing `$(workspaces.<name>.path)`)
+- How will security be reviewed and by whom?
+  - TBD
+- How will UX be reviewed and by whom?
+  - TBD
+
+### Drawbacks
+One potential drawbacks to the proposed solution(s) in the TEP is potential friction users will face
+in migrating their Tekton configuration to the new `.env` format. NOTE: migration is only necessary after
+the deprecation period
+
+## Alternatives
+
+<!--
+What other approaches did you consider and why did you rule them out? These do
+not need to be as detailed as the proposal, but should include enough
+information to express the idea and why it was not acceptable.
+-->
+
+###  Implicitly Project `Parameters` into the Filesystem of `Steps`
+
+Write `Parameters` in files - `/tekton/parameters/<param-name>` - mounted into `Steps`
+in the same way that `Results` are supplied. The file path for the param
+would be made available via a variable (possibly reusing `$(params.<name>)`.
```

Comment:
Files would hold JSON blobs similar to what we do for results I guess?
Producing JSON with bash is not too bad, parsing it is a bit harder, but maybe we could provide some helper functions in a bash resource file mounted in a local folder.

Reply:
+1 to providing some helper scripts to be used for arrays (and possibly objects) as needed. I’ve added a note under the alternative there related to adding UX helper functions